src/quicklist.c

```diff
@@ -924,9 +924,9 @@ REDIS_STATIC void _quicklistInsert(quicklist *quicklist, quicklistEntry *entry,
         new_node->count++;
         quicklistNodeUpdateSz(new_node);
         quicklistRecompressOnly(quicklist, new_node);
-    } else if (full && ((at_tail && node->next && full_next && after) ||
-                        (at_head && node->prev && full_prev && !after))) {
-        /* If we are: full, and our prev/next is full, then:
+    } else if (full && ((at_tail && full_next && after) ||
```

Comment:
@yoav-steinberg The optimization point is here, when using quicklistInsertBefore/quicklistInsertAfter to insert a data at the head and tail of a quicklist, if the current node is full, a new node should be created directly, instead of splitting the current node.